compiler/rustc_hir/src/hir.rs

```diff
@@ -3491,17 +3489,20 @@ pub enum Node<'hir> {
     Arm(&'hir Arm<'hir>),
     Block(&'hir Block<'hir>),
     Local(&'hir Local<'hir>),
-
     /// `Ctor` refers to the constructor of an enum variant or struct. Only tuple or unit variants
     /// with synthesized constructors.
     Ctor(&'hir VariantData<'hir>),
-
     Lifetime(&'hir Lifetime),
     GenericParam(&'hir GenericParam<'hir>),
-
     Crate(&'hir Mod<'hir>),
-
     Infer(&'hir InferArg),
+    WhereBoundPredicate(&'hir WhereBoundPredicate<'hir>),
+    Let(&'hir Let<'hir>),
+    // Span by reference to minimize `Node`'s size
+    #[allow(rustc::pass_by_value)]
+    ArrayLenInfer(&'hir Span),
```

Comment:
```suggestion
    ArrayLenInfer(&'hir ArrayLen),
```
?

Reply:
The second half of `ArrayLen` does not correspond to the id, it has its own it.

This could be `ArrayLenInfer(&'hir ArrayLenInfer)` if the fields of `ArrayLen::Infer` were factored into a separate structure.